### DISCUSSION TITLE
ESP32 플래싱에서 지속적인 에러 발생

### DISCUSSION THREAD

**Discussion by HarinMoon** [luftaquila/monolith#24] [Q&A] [answered]
안녕하세요 고려대학교 어울수레입니다.

저희는 luftaquila님께서 오픈소스로 배포해주신 Monolith 프로젝트 덕분에 부족한 실력을 갖고도 데이터로거를 제작하고 있습니다. 귀한 정보 공유해주셔서 감사한 뜻을 먼저 전하고자 합니다.

 현재 저희는 데이터로거 하드웨어를 제작하였고, TMA-1 Configuration Tool을 이용하여 TMA-1 플래싱, RTC 싱크, ESP32 플래싱을 차례로 하고 있습니다. TMA-1 플래싱과 RTC 싱크는 드라이버 설치 후 정상적으로 이루어졌으나, 정확히 파악하지 못한 이유로 인해 ESP32 플래싱은 같은 메시지가 뜨며 계속 실패하고 있습니다. 

앞서 Discussions에 창원대학교 SKID 팀에서도 비슷한 문제를 겪은 경험을 공유해주셔서 이에 대해 살펴보았습니다. 다만 SKID 팀의 경우 Building은 이루어지고, 마지막 Flashing 단계에서 문제가 발생한 것으로 파악되나, 저희의 경우 Building 단계부터 문제가 발생하여 실패하고 있습니다. 

SKID팀에서 제시해주신 Arduino IDE를 통한 우회 설정도 시도해보았으나 라이브러리 호환성 문제로 인해 우회 플래싱이 불가능했습니다. 

콘솔창에 나타난 메시지를 첨부하겠습니다. 

[Cmonolithbuildermain.exe.txt](https://github.com/user-attachments/files/16285449/Cmonolithbuildermain.exe.txt)

저희가 데이터로깅, 프로그래밍 영역은 초보자 수준의 인원이 공부해가며 준비하고 있어 부족한 부분이 많습니다. 가능하다면 어떤 문제인지, 어떻게 해결할 수 있는지 알려주시면 감사하겠습니다.

**Comment by luftaquila** [comment]
펌웨어와 라이브러리 컴파일은 정상적으로 이루어졌는데, 마지막에 링크하는 중에 심볼을 못 찾아서 undefined reference 에러가 나고 있네요.

링크 에러 나는 심볼 중에 `SocketIOclient` 같은 라이브러리 클래스도 있긴 한데, 아두이노 내장 클래스인 `Serial`이나 하다못해 제일 기본적인 `pinMode`나 `digitalWrite`도 못 찾고 있네요. 뭔가 정상적인 상황은 아닌 것 같은데 저도 일단 새로 처음부터 다시 시도해서 빌드가 정상적으로 되는지 한 번 확인해 보겠습니다.

+) 새로 받아서 시도해 보니 저도 동일한 에러가 발생하네요

**Reply by luftaquila** [reply, marked as answer]
`esp32-arduino` 툴체인이 v3.0.x로 올라가면서 발생하는 문제로 확인되었고, `v2.0.17` 버전으로 pin하여 수정했습니다 (#26). 제 컴퓨터에서는 정상적으로 빌드되네요.

[v1.1.1](https://github.com/luftaquila/monolith/releases/tag/v1.1.1)로 다시 시도해보시고 결과 알려주시면 감사드리겠습니다.

플래싱 후에 stm32, esp32 모두 정상 작동하는지도 알려주시면 좋을 듯 합니다.

**Reply by HarinMoon** [reply]
v.1.1.1로 다시 시도하니 모두 정상적으로 플래싱 되었고, 작동합니다. 빠른 피드백 감사합니다.
